spark/src/main/scala/geotrellis/spark/io/avro/codecs/TileCodecs.scala

```diff
@@ -32,7 +35,9 @@ trait TileCodecs {
         .asScala // notice that Avro does not have native support for Short primitive
         .map(_.toShort)
         .toArray
-      new ShortArrayTile(array, rec[Int]("cols"), rec[Int]("rows"))
+      val cellType = CellType.fromString(rec[Utf8]("celltype").toString)
```

Comment:
A couple issues:
1. the enum doesn't seem like it would be capable of supporting the user defined case (we'd need to send the user defined nodata in addition to the enum)
2. the optional nodata approach breaks backward compatibility since a default would prevent us from encoding raw tiles